### ISSUE TITLE
Separate  Model dan DTO

### ISSUE THREAD

**Issue by mhdiiilham** [nandonyata/go-gorm#1]
https://github.com/nandonyata/go-gorm/blob/fa2657371390f1f24fd362f896efbfe0d15feeb6/models/user.go#L6

Embedding the `gorm.Model` inside your model is not a good design, tho, therefore you should be separating the model and the DTO's (the data type that you're using to interact with the database).

I recommend you read this [The Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html).

In short, your inner layer (entity or model) shouldn't depend on any external dependencies.

The response model should also not in the model layer. Most simple way is to declare this type at `Controller` package
https://github.com/nandonyata/go-gorm/blob/fa2657371390f1f24fd362f896efbfe0d15feeb6/models/user.go#L12
